router/dashboard.go

```diff
@@ -2,30 +2,52 @@ package router
 
 import (
 	"budget-helper/database"
+	"errors"
+	"fmt"
 	"log"
 	"net/http"
+	"strconv"
+	"strings"
+
+	"gorm.io/gorm"
 )
 
 // -- DASHBOARD & MAIN APP ROUTES --
 func (rt *Router) handleDashboard(w http.ResponseWriter, r *http.Request) {
 	data := struct {
-		AppTitle        string
-		PageTitle       string
-		Budget          *database.Budget
-		Categories      []database.Category
-		BudgetRemaining float64
-		BufferRemaining float64
-		PercentageSpent int
+		AppTitle         string
+		PageTitle        string
+		Budget           *database.Budget
+		Categories       []database.Category
+		BalanceRemaining string
+		BufferRemaining  string
+		PercentageSpent  int
 	}{
 		AppTitle:  AppTitle,
 		PageTitle: "Dashboard",
 	}
 
+	getBudgetID := strings.TrimSpace(r.URL.Query().Get("id"))
+	id, err := strconv.Atoi(getBudgetID)
+	if err != nil {
+		displayErrorPage(w, r, http.StatusBadRequest, "Bad Request",
```

Comment:
Should probably omit the string input `"Bad Request"` and make the `displayErrorPage` figure it out based on the status code.